src/storage/backend/backend_impl.rs

```diff
@@ -1,19 +1,28 @@
 //! The backend implementation.
 
+use std::time::Duration;
+
 use async_trait::async_trait;
 use clippy_utilities::OverflowArithmetic;
 use datenlord::config::{StorageParams, StorageS3Config};
 use datenlord::metrics::DATENLORD_REGISTRY;
 use futures::{stream, AsyncReadExt, AsyncWriteExt, StreamExt};
-use opendal::layers::PrometheusLayer;
+use opendal::layers::{ConcurrentLimitLayer, PrometheusLayer, RetryLayer};
 use opendal::services::{Fs, S3};
 use opendal::{ErrorKind, Operator};
 use prometheus::{exponential_buckets, linear_buckets};
+use tokio::time::sleep;
+use tracing::debug;
 
 use crate::async_fuse::fuse::protocol::INum;
 use crate::storage::error::StorageResult;
 use crate::storage::{Block, Storage};
 
+/// The maximum number of retries for writing a block.
+const MAX_RETRIES: usize = 3;
+/// The delay between retries for writing a block.
+const RETRY_DELAY: Duration = Duration::from_secs(1);
```

Comment:
Does it really need to sleep for 1s?

Reply:
It's temporary sleep, we will change to `work pool` in the plan.